src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala

```diff
@@ -97,10 +47,118 @@ object TestUtils {
     printDf(expected)
     println("   symmetric Difference ")
     symmetricDifference(actual)(expected)
-      .withColumn("_df", when($"_df" === 1,"actual").otherwise("expected"))
+      .withColumn("_df", when($"_in_first_df","actual").otherwise("expected"))
+      .drop($"_in_first_df")
       .show(false)
   }
 
   def printFailedTestResult(testName: String, argument: DataFrame)(actual: DataFrame)(expected: DataFrame): Unit = printFailedTestResult(testName, Seq(argument))(actual)(expected)
 
+  def testArgumentExpectedMapWithComment[K,V](experiendum: K=>V)(argExpMapComm: Map[(String,K),V]): Unit = {
+    def logFailure(argument: K, actual:V, expected: V, comment: String): Unit = {
+      logger.error("Test case failed !")
+      logger.error(s"   argument = $argument")
+      logger.error(s"   actual   = $actual")
+      logger.error(s"   expected = $expected")
+      logger.error(s"   comment  = $comment")
+    }
+    def checkKey(x: (String,K)): Boolean = x match {
+      case (comment, argument) =>
+        val actual = experiendum(argument)
+        val expected = argExpMapComm(x)
+        val resultat = actual == expected
+        if (!resultat) logFailure(argument, actual, expected, comment)
+        resultat
+      case _ => throw new Exception(s"Something went wrong: checkKey called with parameter x=$x")
+    }
+    val results: Set[Boolean] = argExpMapComm.keySet.map(checkKey)
+    assert(results.forall(p=>p))
+  }
+
+  def testArgumentExpectedMap[K,V](experiendum: K=>V)(argExpMap: Map[K,V]): Unit = {
```

Comment:
```suggestion
  def testArgumentExpectedMap[K,V](experiendum: K=>V, argExpMap: Map[K,V]): Unit = {
```
If no explicit need, use only one argument list.

Reply:
Done: https://github.com/zzeekk/spark-temporalquery/pull/11/commits/26ee7ae49877695da4343f997eff239aa7e73989